_posts/2018-08-30-on-the-fly-transformations.md

```diff
@@ -0,0 +1,283 @@
+Trajectory transformations in MDAnalysis, on the other end, have one great advantage - they are performed on-the-fly. This means that, after loading the trajectory file, the user adds a transformation workflow and the transformations are applied to each frame that is read. There is no need to iterate over the whole trajectory before performing other analysis, and, with the frame indexing provided by MDAnalysis, only the frames of interest are processed. Moreover, the way the transformations API is implemented makes it really easy to add custom transformations.
+Another things that really makes the "on-the-fly" aspect of the MDAnalysis transformations really shine is coupling it to a visualization widget such as [NGL Viewer](http://nglviewer.org/ngl/api/index.html).
+
+## Using MDAnalysis transformations
+Now it's time to learn how to use the trajectory transformations in MDAnalysis. During the following steps, we will apply some transformations on a 1 ns trajectory of a simple 19-residue peptide embeded in a 128-DMPC membrane, showing the GROMACS `trjconv` command and the equivalent MDAnalysis code and output. To keep thinks lightweight, frames are were taken every 100 ps, and water molecules were removed.
+
+### Example 1: making everything whole again
+When performing MD simulations using periodic boundary conditions, molecules will often cross the limits of the unit cell. When this happens, some atoms of the molecule will show up on the the opposing side of the unit cell and some molecular viewers will show stretched bonds and other visual artifacts depending on the visual representation of the system. 
+This is the case of our system. Without any modifications, when we look at the trajectory of our system, things will become very ugly, very fast:
+
+
+```python
+import warnings
+warnings.filterwarnings('ignore') # some attributes are missing 
+import MDAnalysis as mda
+import nglview as nv
+u = mda.Universe('pept_in_memb.tpr', 'pept_in_memb.xtc')
+nv.show_mdanalysis(u)
+```
+
+
+    NGLWidget(count=11)
+
+
+Using `trjconv`, one way to may every molecules whole again would be:
+
+    gmx trjconv -f pept_in_memb.xtc -s pept_in_memb.tpr -pbc mol -o output.xtc
+    
+In MDAnalysis this can be done with the `unwrap` transformation, which takes an AtomGroup as argument. This can be done as follows:
+
+
+```python
+# a custom atom group can be passed as an argument. In this case we will use all the atoms
+# in the Universe u
+u = mda.Universe('pept_in_memb.tpr', 'pept_in_memb.xtc')
+ag = u.atoms
+# we define the transformation
+transformation = mda.transformations.unwrap(ag)
+# we add the transformation to the trajectory in it will be applied everytime we read a frame
+u.trajectory.add_transformations(transformation)
+# we can do other things with the trajectory without having to generate a file with the transformed
```

Comment:
This should be it's own paragraph separate from this and not hidden in a comment.